nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-3-service/src/main/java/org/apache/nifi/kafka/service/Kafka3ConnectionService.java

```diff
@@ -249,15 +252,33 @@ public class Kafka3ConnectionService extends AbstractControllerService implement
     private volatile ServiceConfiguration serviceConfiguration;
     private volatile Properties producerProperties;
     private volatile Properties consumerProperties;
+    private volatile String uri;
 
     @OnEnabled
     public void onEnabled(final ConfigurationContext configurationContext) {
         clientProperties = getClientProperties(configurationContext);
         serviceConfiguration = getServiceConfiguration(configurationContext);
         producerProperties = getProducerProperties(configurationContext, clientProperties);
         consumerProperties = getConsumerProperties(configurationContext, clientProperties);
+        uri = createBrokerUri(configurationContext);
     }
 
+    private String createBrokerUri(final ConfigurationContext context) {
+        final String bootstrapServers = context.getProperty(BOOTSTRAP_SERVERS).getValue();
+        final String[] bootstrapServersArray = bootstrapServers.split(",");
+        String firstBootstrapServer = bootstrapServersArray[0].trim();
+        if (firstBootstrapServer.contains("://")) {
+            firstBootstrapServer = firstBootstrapServer.substring(firstBootstrapServer.indexOf("://") + 3);
```

Comment:
I am sorry for my ignorance here regarding Kafka configuration but will all the configured servers always have `://`. The way I read th Kafka documentation `host:port` where host may not include the leading `://`.